behat-tests/features/bootstrap/FeatureContext.php

```diff
@@ -197,6 +225,8 @@ public function createPersonalizedElements(TableNode $elementsTable) {
    * @Then I should see :count for the :type count
    */
   public function assertMenuCount($count, $type) {
+    $this->spinUntilAjaxIsFinished();
```

Comment:
Haha I will be very happy to try to convince you on that.

#### Why "I wait for AJAX to finish" is evil
We are going through all the trouble implementing and maintaining this very expensive Gherkin language is only because we are now writing an non-tech readable article instead of a program. 

This means, at the end game of a complete Behat Suite, we are going to be almost reading an series of articles that feel very much not techie, in fact, so plain that your newbie-dev, QA, PM, customer, marketer, stakeholders, and average internet surfers are able to understand what this product is doing, and they do so without calling customer service department or reading the code. And it is a live documentation so everybody can count on it.

For this reason, techie language such as *I wait for AJAX to finish*, *I wait for 7 seconds*, *I click on the link ".some-element-class"*, *I see the page response code is "200"*, *I practice some voodoo that you-don't-know* are counter productive to the use of Gherkin/Behat, therefore defeating the most important reason why we are using it.

The best reference I know for this topic is this [talk](http://everzet.com/post/33359584167/video-of-my-talk-from-symfony-live-2012-paris) by everzet himself. I checked on him about 2 month ago and he is still saying the same thing so I think we are good there. =D

#### What's the cost of assertion?
The cost of assertion, JavaScript or not, is literally in milliseconds. 

(Stay tuned. To be continued ...)